Portal/src/Datahub.Infrastructure/Services/ResourceMessagingService.cs

```diff
@@ -39,6 +39,28 @@ public async Task SendToTerraformDeleteQueue(WorkspaceDefinition project, int pr
         }
     }
 
+    public async Task SendToTerraformDeleteResourceQueue(WorkspaceDefinition project, int projectId, string tool)
+    {
+        // deleting resources from toolbox
+
+        await sendEndpointProvider.SendDatahubServiceBusMessage(QueueConstants.ResourceDeleteRequestQueueName, project);
+        
+        await using var ctx = await dbContextFactory.CreateDbContextAsync();
+
+        var projectChanged = await ctx.Projects
+            .FirstOrDefaultAsync(p => p.Project_ID == projectId);
+
+        var resourceToDeleted = await ctx.Project_Resources2
+            .FirstOrDefaultAsync(r => r.ResourceType.EndsWith(tool));
+
+        if (resourceToDeleted != null && projectChanged != null)
+        {
+            resourceToDeleted.UpdatedAt = DateTime.UtcNow;
+            projectChanged.Resources.Remove(resourceToDeleted);
```

Comment:
We don't want to actually delete the database record of it here but rather do a "soft delete" instead.

Either way it's outside the scope of this PR but just a heads up! 😄